### DISCUSSION TITLE
How to set a language on a semantic token

### DISCUSSION THREAD

**Discussion by ggrossetie** [microsoft/vscode-discussions#869] [Extension Development QnA] [answered]
The documentation states (emphasis mine):

> Each token returned by the provider comes with a classification that consists of a token type, any number of token modifiers, **and a token language**.

> The output of a semantic token provider consists of tokens. Each token has a range and a token classification that describes what kind of syntax element the token represents. **Optionally, the classification can also name a language, if the token is part of an embedded language.**

https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide

I'm working on the AsciiDoc extension and the AsciiDoc language allows defining code blocks (just like in Markdown). As a result, I would like to create semantic tokens where the language is the language of the code block.

```
[source,js]
----
// will print hello
console.log('hello')
----
```

My goal is to return a semantic token that contains the whole code block with the language JavaScript (js):

```js
// will print hello
console.log('hello')
```

I don't really want/need to specify a token type or a token modifier since I want to delegate all the work to the JavaScript semantic highlighter.

I can't find how to do it... is it possible?

**Comment by tamuratak** [comment, marked as answer]
Not supported. The feature request is tracked at microsoft/vscode/issues/163292.

You would be interested in https://github.com/microsoft/vscode-anycode.

**Reply by ggrossetie** [reply]
Thanks!
I wasn't able to find this issue. Maybe we should update the documentation to mention that this is not yet supported?
